crates/re_viewer/src/app.rs

```diff
@@ -562,6 +566,33 @@ impl App {
         }
     }
 
+    #[cfg(target_arch = "wasm32")]
+    fn run_copy_direct_link_command(&mut self, store_context: Option<&StoreContext<'_>>) {
+        let Some(SmartChannelSource::RrdHttpStream { url }) = store_context
+            .and_then(|ctx| ctx.recording)
+            .and_then(|rec| rec.data_source.as_ref())
+        else {
+            self.toasts.add(toasts::Toast {
+                kind: toasts::ToastKind::Warning,
+                text: format!("Could not copy direct link, no recording is open"),
+                options: toasts::ToastOptions::with_ttl_in_seconds(4.0),
+            });
+            return;
+        };
+        let direct_link = match &self.startup_options.location {
```

Comment:
When is `StartupOptions::location` `None`? Looking at the code it seems to always be set on wasm, so maybe it doesn't need to be an `Option`? In any case, you could also call `eframe::web::web_location()` here if you prefer. That way we don't need the hard-coded fallback to `app.rerun.io`

Reply:
Looks like it's there because of this `Default` impl:

https://github.com/rerun-io/rerun/blob/2969b83048439226f8e4858b9a1e42bfa7573bd4/crates/re_viewer/src/app.rs#L70